src/NUnitFramework/framework/Constraints/Comparers/EquatablesComparer.cs

```diff
@@ -61,35 +61,53 @@ internal EquatablesComparer(NUnitEqualityComparer equalityComparer)
 
         private static MethodInfo FirstImplementsIEquatableOfSecond(Type first, Type second)
         {
-            var pair = new KeyValuePair<Type, MethodInfo>();
+            var mostDerived = default(EquatableMethodImpl);
 
-            foreach (var xEquatableArgument in GetEquatableGenericArguments(first))
-                if (xEquatableArgument.Key.IsAssignableFrom(second))
-                    if (pair.Key == null || pair.Key.IsAssignableFrom(xEquatableArgument.Key))
```

Comment:
Original PR used KeyValuePair<> for storing all this. During development I created a custom struct because an earlier approach required a third piece of data to store. I eventually got it working with only `Type` and `MethodInfo` again, but I kept the custom struct since I found it easier to read this way.

Please let me know if you'd like me to revert it back to using `KeyValuePair<Type, MethodInfo>`.